lib/gmail/message_attachment.ex

```diff
@@ -0,0 +1,58 @@
+defmodule Gmail.MessageAttachment do
+
+  @moduledoc """
+  An email message attachment.
+  """
+
+  alias __MODULE__
+  alias Gmail.{Helper}
+  import Gmail.Base
+
+  @doc """
+  Gmail API documentation: https://developers.google.com/gmail/api/v1/reference/users/messages/attachments
+  """
+  defstruct attachmentId: "",
+    size: "",
+    data: ""
+
+  @type t :: %__MODULE__{}
+
+  @doc """
+  Gets the specified attachment.
+
+  Gmail API documentation: https://developers.google.com/gmail/api/v1/reference/users/messages/attachments/get
+  """
+  @spec get(String.t, String.t, String.t) :: {atom, String.t, String.t}
+  def get(user_id, message_id, id) do
+    path = querify_params("users/#{user_id}/messages/#{message_id}/attachments/#{id}", [], %{})
+
+    {:get, base_url, path}
+  end
+
+  @doc """
+  Converts a Gmail API attachment resource into a local struct.
+  """
+  @spec convert(map) :: MessageAttachment.t
+  def convert(message) do
+    attachment = message
+    |> Helper.atomise_keys
+
+    attachment = struct(MessageAttachment, attachment)
+  end
+
+  @doc """
+  Handles an attachment resource response from the Gmail API.
+  """
+  def handle_attachment_response(response) do
+    response
+    |> handle_error
+    |> case do
+      {:error, message} ->
+        {:error, message}
+      {:ok, raw_message} ->
+        {:ok, MessageAttachment.convert(raw_message)}
+      # :ok -> :ok # uh... i'm not sure why i should handle this case.
```

Comment:
@craigp while I was wiring up tests, I hit this case when using empty strings for message_id and attachment_id. Should I handle the case? I believe it's from `do_parse_response` ln:66 of http.ex